qt/widgets/common/src/MantidWSIndexDialog.cpp

```diff
@@ -628,34 +629,53 @@ void MantidWSIndexWidget::populateLogComboBox() {
   // First item should be "Workspace index"
   m_logSelector->addItem(WORKSPACE_NAME);
 
-  // Create a table of all single-value numeric log names versus
-  // how many workspaces they appear in
-  std::map<std::string, int> logCounts;
+  // Create a map of all logs and their double represenation to compare against.
+  // Only logs that can be converted to a double and are not all equal will make
+  // the final cut
+  // it is map[logname] = (isconstantvalue, value)
+  std::map<std::string, std::pair<bool, double>> usableLogs;
```

Comment:
Do you think a struct might be more clear than `std::pair<bool, double>`? Something like 
```c++
struct TestStruct {
  TestStruct(bool isconstantvalue, double value)
      : isconstantvalue(isconstantvalue), value(value) {}
  bool isconstantvalue;
  double value;
};
```
changes things like `logItem.second.first = (value == logItem.second.second)` to `logItem.second.isconstantvalue = (value == logItem.second.value)` which I think is at least _slightly_ more readable